File: RealEstateAgencyC++/RealEstateAgency/Property.cpp
#include "Property.h"
#include <iostream>
#include <string>

using namespace std;

Property::Property() {
	strcpy(address, "");
	strcpy(owner, "");
	price = 0;
	area = 0;
}

Property::Property(const char* address, const char* owner, const double price, const int area) {
	strcpy_s(this->address, address);
	strcpy_s(this->owner, owner);
	this->price = price;
	this->area = area;
}

Property::Property(Property& pr) {
	strcpy_s(this->address, pr.address);
	strcpy_s(this->owner, pr.owner);
	this->price = pr.price;
	this->area = pr.area;
}

Property::~Property(){}

Property& Property::operator=(const Property& pr) {
	strcpy_s(this->address, pr.address);
	strcpy_s(this->owner, pr.owner);
	this->price = pr.price;
	this->area = pr.area;
	return *this;
}

char* Property::GetAddress() {
	return address;
}

double Property::GetPrice() {
	return price;
}

void Property::setProperty(const char* address, const char* owner, const double price, const int area) {
	strcpy(this->address, address);
	strcpy(this->owner, owner);
	this->price = price;
	this->area = area;
}

void Property::PropertyPrint() {
	cout << "Address: " << address << endl;
	cout << "Owner: " << owner << endl;
	cout << "Price: " << price << endl;
	cout << "Area: " << area << endl;
}

File: RealEstateAgencyCS/RealEstateAgencyCS/Property.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RealEstateAgencyCS
{
    class Property
    {
        protected string address;
        protected string owner;
        protected double price;
        protected int area;

        public Property()
        {
            address = "";
            owner = "";
            price = 0.00;
            area = 0;
        }
        
        public Property(string address, string owner, double price, int area)
        {
            this.address = address;
            this.owner = owner;
            this.price = price;
            this.area = area;

        }

        public Property(Property pr)
        {
            this.address = pr.address;
            this.owner = pr.owner;
            this.price = pr.price;
            this.area = pr.area;
        }

        virtual public string GetAddress()
        {
            return address;
        }

        virtual public double GetPrice()
        {
            return price;
        }

        public void SetPriority(string address, string owner, double price, int area)
        {
            this.address = address;
            this.owner = owner;
            this.price = price;
            this.area = area;
        }

        virtual public void PropertyPrint()
        {
            Console.WriteLine($"Address: {address}");
            Console.WriteLine($"Owner: {owner}");
            Console.WriteLine($"Price: {price}");
            Console.WriteLine($"Area: {area}");
        }

        


            
    }
}
